### PR TITLE
ci(release): Use action-prepare-release and rework dry-run

### PR DESCRIPTION
**Pull request by BYK**
Switches to using getsentry/action-prepare-release for common steps and removes the dry run option from the prepare stage as it prevents us from testing the publish step. Prepare step should be safe anyways and can be skipped with the `skip-prepare` input.

Related: https://app.asana.com/0/1198192131329257/1198192131329301, https://app.asana.com/0/1198192131329257/1198192573081717

#skip-changelog

### CODE REVIEW

**Review by chadwhitacre — APPROVED** [review]
Seems to work:

https://github.com/getsentry/relay/runs/1493511706?check_suite_focus=true

Craft Prepare fails since it's not on `master`. I'd like to find a way to be able to run actions to completion in testing, but for this PR we mostly want to test the next action-prepare-release usage. Not a lot of logging but seems like it worked? I tested the manual version override.

**Review by untitaker — APPROVED** [review]
rubberstamping... better wait for jan if you want a proper review

**Comment by BYK** [comment]
>rubberstamping... better wait for jan if you want a proper review

This is already reviewed by @chadwhitacre, just trying to unblock myself for landing. I think I'll change CODEOWNERS for this particular workflow.

**Pull request merged**
